pageserver/src/config.rs

```diff
@@ -1019,7 +427,14 @@ impl PageServerConf {
             );
         }
 
-        conf.default_tenant_conf = t_conf.merge(TenantConf::default());
+        IndexEntry::validate_checkpoint_distance(conf.default_tenant_conf.checkpoint_distance)
+            .map_err(|msg| anyhow::anyhow!("{msg}"))
```

Comment:
I am a bit unsure what does this do, but it looks weird. If we really have a `Result<(), &'static str>` returning fn, then `anyhow::Error::msg`
```suggestion
            .map_err(anyhow::Error::msg)
```

Reply:
b598782b430b4f301a9305b577e0771ca7f811fb